ls1intum/Artemis
# Change landing page for LTI users to new exercise details

### Checklist

### Motivation and Context
MOOC users (e.g. edx) have still been using the old courses page which was outdated and did not work any more properly. We now redirect them to the new exercise details page which makes much more sense. 
We can now finally remove the old courses and exercise list and resolve #851 

### Description
Refactored the redirect to the landing page via adapting URL in LTIResource, moved some code (e.g. Welcome Alert to the exercise details page) and removed a lot of old code.
The only class that was still actively used in the old course list module was the course score calculation service. I moved it into the overview module and fixed some compilation errors in the jest tests.

There are no UI changes.

### Steps for Testing
This is rather hard to test for others, because we need to adapt the LTI URL defined in our MOOC to the test server. I will make sure this works on the test server, before we deploy it.
